pex/resolver.py

```diff
@@ -151,10 +151,11 @@ def filter_packages_by_interpreter(cls, packages, interpreter, platform):
     return [package for package in packages
         if package.compatible(interpreter.identity, platform)]
 
-  def __init__(self, allow_prereleases=None, interpreter=None, platform=None):
+  def __init__(self, allow_prereleases=None, interpreter=None, platform=None, pkg_blacklist=None):
     self._interpreter = interpreter or PythonInterpreter.get()
     self._platform = platform or Platform.current()
     self._allow_prereleases = allow_prereleases
+    self._blacklist = pkg_blacklist or {}
```

Comment:
Not sure if it matters here.  This line carries risk of `pkg_blacklist` being updated outside of the function and impact execution inside since pkg_blacklist becomes a shared object.  Maybe `self._blacklist = {}.update(pkg_blacklsit or {})` will be safer?